tdrs-backend/tdpservice/parsers/util.py

```diff
@@ -86,3 +87,221 @@ def contains_encrypted_indicator(line, encryption_field):
+    elif query == "models":
+        links = schema_options.get(program, {}).get(section, None)
+
+        # if query is not chosen or wrong input, return all options
+        # query = 'models', model = 'T1'
+        models = links.get(query, links)
+
+        if model_name is None:
+            return models
+        elif model_name not in models.keys():
+            return None  # intentionally trigger the error_msg for unknown record type
+        else:
+            return models.get(model_name, models)
+
+
+'''
+text -> section YES
+text -> models{} YES
+text -> model YES
+datafile -> model
+    ^ section -> program -> model
+datafile -> text
+model -> text YES
+section -> text
+
+text**: input string from the header/file
+'''
+
+def get_program_models(str_prog, str_section):
+    """Return the models dict for a given program and section."""
+    return get_schema_options(program=str_prog, section=str_section, query='models')
+
+def get_program_model(str_prog, str_section, str_model):
+    """Return singular model for a given program, section, and name."""
+    return get_schema_options(program=str_prog, section=str_section, query='models', model_name=str_model)
+
+def get_section_reference(str_prog, str_section):
+    """Return the named section reference for a given program and section."""
+    return get_schema_options(program=str_prog, section=str_section, query='section')
+
+def get_text_from_df(df):
+    """Return the short-hand text for program, section for a given datafile."""
+    return get_schema_options("", section=df.section, query='text')
+
+def get_prog_from_section(str_section):
+    """Return the program type for a given section."""
+    # e.g., 'SSP Closed Case Data'
+    if str_section.startswith('SSP'):
+        return 'SSP'
+    elif str_section.startswith('Tribal'):
+        return 'TAN'  # problematic, do we need to infer tribal entirely from tribe/fips code?
+    else:
+        return 'TAN'
+
+    # TODO: if given a datafile (section), we can reverse back to the program b/c the
+    # section string has "tribal/ssp" in it, then process of elimination we have tanf
+
+def get_schema(line, section, program_type):
+    """Return the appropriate schema for the line."""
+    line_type = line[0:2]
+    return get_schema_options(program_type, section, query='models', model_name=line_type)
+
+def fiscal_to_calendar(year, fiscal_quarter):
+    """Decrement the input quarter text by one."""
+    array = [1, 2, 3, 4]  # wrapping around an array
+    int_qtr = int(fiscal_quarter[1:])  # remove the 'Q', e.g., 'Q1' -> '1'
+    if int_qtr == 1:
+        year = year - 1
+
+    ind_qtr = array.index(int_qtr)  # get the index so we can easily wrap-around end of array
+    return year, "Q{}".format(array[ind_qtr - 1])  # return the previous quarter
+
+
+def transform_to_months(quarter):
+    """Return a list of months in a quarter."""
+    match quarter:
+        case "Q1":
+            return ["Jan", "Feb", "Mar"]
```

Comment:
I believe your objection is handled via `fiscal_to_calendar`. No?

Reply:
i missed that, thanks. i do feel it's much more straightforward to hard-code the expected fiscal months in `transform_to_months`, though, since that's how it's used elsewhere in the system (emails, data file submission frontend). are we going to switch between the calendar and fiscal quarters?